windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -184,7 +184,37 @@ UwbDeviceConnector::SessionInitialize(uint32_t sessionId, UwbSessionType session
 {
     std::promise<UwbStatus> resultPromise;
     auto resultFuture = resultPromise.get_future();
-    // TODO: invoke IOCTL_UWB_SESSION_INIT
+
+    wil::shared_hfile handleDriver;
+    auto hr = OpenDriverHandle(handleDriver, m_deviceName.c_str());
+    if (FAILED(hr)) {
+        PLOG_ERROR << "failed to obtain driver handle for " << m_deviceName << ", hr=" << hr;
+        return resultFuture;
+    }
+
+    const UWB_SESSION_INIT sessionInit{
+        .size = sizeof(UWB_SESSION_INIT),
+        .sessionId = sessionId,
+        .sessionType = UwbCxDdi::From(sessionType),
+    };
+    UWB_STATUS status;
+
+    // Determine the amount of memory required for the UWB_SESSION_INIT from the driver.
+    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_SESSION_INIT, const_cast<UWB_SESSION_INIT*>(&sessionInit), sizeof sessionInit, &status, sizeof status, nullptr, nullptr);
+    if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
+        // TODO: need to do something different here

```

Comment:
This should set an exception in resultPromise (as is done below). The convention we've used so far for errors incurred before the driver gets hit (like here) is to provide UwbStatusGeneric::Rejected as the promise's exception, eg.



```suggestion
        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
```
 Same goes for all error cases where the desired return value isn't valid (exception set in all cases except success).